server/src/org/apache/cloudstack/ca/CAManagerImpl.java

```diff
@@ -358,6 +358,16 @@ protected void runInContext() {
                                             hostDescription, certificate.getNotAfter()));
                             alertMap.put(hostIp, new Date());
                         }
+                    } else if (now.isAfter(warnDate)) {
+                        if (alertMap.containsKey(hostIp)) {
+                            final Date lastSentDate = alertMap.get(hostIp);
+                            if (now.minusDays(1).toDate().before(lastSentDate)) {
+                                continue;
+                            }
+                        }
+                        caManager.sendAlert(host, "Certificate expiring soon for " + hostDescription,
+                                String.format("Certificate is going to expire for %s on %s. It will auto renew in %s days before that.", hostDescription, certificate.getNotAfter(), CertExpiryAlertPeriod.valueIn(host.getClusterId())));
```

Comment:
This could be misleading.  The cert may not auto-renew if `AutomaticCertRenewal.valueIn(host.getClusterId())` is not `true`

Maybe an if statement or something to state that it will "auto-renew" after %s days or state that auto-renewing is not enabled?

Reply:
I've expanded the alert messaging to cover all the different scenarios. Let me know if that works for you.